platform/ios/src/MGLMapboxEvents.m

```diff
@@ -880,7 +930,133 @@ - (void)URLSession:(NSURLSession *)session didReceiveChallenge:(NSURLAuthenticat
+
+            NSFileManager *fileManager = [[NSFileManager alloc] init];
+            if ([fileManager fileExistsAtPath:logFilePath]) {
+                //NSLog(@"file handle'ing %@", event[@"event"]);
+                NSFileHandle *fileHandle = [NSFileHandle fileHandleForWritingAtPath:logFilePath];
+                [fileHandle seekToEndOfFile];
+                [fileHandle writeData:[jsonString dataUsingEncoding:NSUTF8StringEncoding]];
+            } else {
+                //NSLog(@"writing new log file %@", event[@"event"]);
+                [fileManager createFileAtPath:logFilePath contents:[jsonString dataUsingEncoding:NSUTF8StringEncoding] attributes:@{ NSFileProtectionKey: NSFileProtectionCompleteUntilFirstUserAuthentication }];
+            }
+        }
+
+    });
+}
+
+- (NSString *) stringForDebugEvent:(MGLMapboxEventAttributes *)event {
+    // redact potentially sensitive location details from system console log
+    if ([event[@"event"] isEqualToString:MGLEventTypeLocation] ||
+        [event[@"event"] isEqualToString:MGLEventTypeVisit]) {
+        MGLMutableMapboxEventAttributes *evt = [MGLMutableMapboxEventAttributes dictionaryWithDictionary:event];
+        [evt setObject:@"<redacted>" forKey:@"lat"];
+        [evt setObject:@"<redacted>" forKey:@"lng"];
+        event = evt;
+    }
+
+    return [NSString stringWithFormat:@"Mapbox Telemetry event %@", event];
+}
+
+- (BOOL) isProbablyAppStoreBuild {
+#if TARGET_IPHONE_SIMULATOR
+    return NO;
+#else
+    // BugshotKit by Marco Arment https://github.com/marcoarment/BugshotKit/
+    // Adapted from https://github.com/blindsightcorp/BSMobileProvision
+
+    NSString *binaryMobileProvision = [NSString stringWithContentsOfFile:[NSBundle.mainBundle pathForResource:@"embedded" ofType:@"mobileprovision"] encoding:NSISOLatin1StringEncoding error:NULL];
+    if (! binaryMobileProvision) return YES; // no provision
+
+    NSScanner *scanner = [NSScanner scannerWithString:binaryMobileProvision];
+    NSString *plistString;
+    if (! [scanner scanUpToString:@"<plist" intoString:nil] || ! [scanner scanUpToString:@"</plist>" intoString:&plistString]) return YES; // no XML plist found in provision
+    plistString = [plistString stringByAppendingString:@"</plist>"];
+
+    NSData *plistdata_latin1 = [plistString dataUsingEncoding:NSISOLatin1StringEncoding];
+    NSError *error = nil;
+    NSDictionary *mobileProvision = [NSPropertyListSerialization propertyListWithData:plistdata_latin1 options:NSPropertyListImmutable format:NULL error:&error];
+    if (error) return YES; // unknown plist format
+
+    if (! mobileProvision || ! mobileProvision.count) return YES; // no entitlements
+
+    if (mobileProvision[@"ProvisionsAllDevices"]) return NO; // enterprise provisioning
+
+    if (mobileProvision[@"ProvisionedDevices"] && ((NSDictionary *)mobileProvision[@"ProvisionedDevices"]).count) return NO; // development or ad-hoc
```

Comment:
Nit: You can avoid this ugly cast by using the bracketed message syntax: `[mobileProvision[@"ProvisionedDevices"] count]`.